core/drivers/crypto/hisilicon/hisi_qm.c

```diff
@@ -0,0 +1,791 @@
+		EMSG("QM mailbox operation timeout");
+		return HISI_QM_DRVCRYPT_EBUSY;
+	}
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static void qm_cfg_vft_data(struct hisi_qm *qm, uint8_t vft_type,
+			    uint32_t base, uint32_t number)
+{
+	uint32_t data_h = 0;
+	uint32_t data_l = 0;
+	uint64_t data = 0;
+
+	switch (vft_type) {
+	case QM_SQC_VFT:
+		data = SHIFT_U64(base, QM_SQC_VFT_START_SQN_SHIFT) |
+			QM_SQC_VFT_VALID |
+			SHIFT_U64((number - 1), QM_SQC_VFT_SQ_NUM_SHIFT);
+		break;
+	case QM_CQC_VFT:
+		data = QM_CQC_VFT_VALID;
+		break;
+	default:
+		panic("Invalid vft type");
+	}
+
+	reg_pair_from_64(data, &data_h, &data_l);
+	io_write32(qm->io_base + QM_VFT_CFG_DATA_L, data_l);
+	io_write32(qm->io_base + QM_VFT_CFG_DATA_H, data_h);
+}
+
+static enum hisi_drv_status qm_set_vft_common(struct hisi_qm *qm,
+					      uint8_t vft_type,
+					      uint32_t function,
+					      uint32_t base,
+					      uint32_t num)
+{
+	uint32_t val = 0;
+
+	if (IO_READ32_POLL_TIMEOUT(qm->io_base + QM_VFT_CFG_RDY, val,
+				   val & QM_FVT_CFG_RDY_BIT, POLL_PERIOD,
+				   POLL_TIMEOUT)) {
+		EMSG("QM VFT is not ready");
+		return HISI_QM_DRVCRYPT_EBUSY;
+	}
+
+	io_write32(qm->io_base + QM_VFT_CFG_OP_WR, QM_VFT_WRITE);
+	io_write32(qm->io_base + QM_VFT_CFG_TYPE, vft_type);
+	io_write32(qm->io_base + QM_VFT_CFG_ADDRESS, function);
+	qm_cfg_vft_data(qm, vft_type, base, num);
+	io_write32(qm->io_base + QM_VFT_CFG_RDY, 0x0);
+	io_write32(qm->io_base + QM_VFT_CFG_OP_ENABLE, QM_FVT_CFG_RDY_BIT);
+
+	if (IO_READ32_POLL_TIMEOUT(qm->io_base + QM_VFT_CFG_RDY, val,
+				   val & QM_FVT_CFG_RDY_BIT, POLL_PERIOD,
+				   POLL_TIMEOUT)) {
+		EMSG("QM VFT is not ready");
+		return HISI_QM_DRVCRYPT_EBUSY;
+	}
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static enum hisi_drv_status qm_set_xqc_vft(struct hisi_qm *qm,
+					   uint32_t function,
+					   uint32_t base, uint32_t num)
+{
+	enum hisi_drv_status ret = HISI_QM_DRVCRYPT_NO_ERR;
+	int i = 0;
+
+	if (!num) {
+		EMSG("Invalid sq num");
+		return HISI_QM_DRVCRYPT_EINVAL;
+	}
+
+	for (i = QM_SQC_VFT; i <= QM_CQC_VFT; i++) {
+		ret = qm_set_vft_common(qm, i, function, base, num);
+		if (ret) {
+			EMSG("QM set type%d fail", i);
+			return ret;
+		}
+	}
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static enum hisi_drv_status qm_get_vft(struct hisi_qm *qm, uint32_t *base,
+				       uint32_t *num)
+{
+	enum hisi_drv_status ret = HISI_QM_DRVCRYPT_NO_ERR;
+	uint64_t sqc_vft = 0;
+
+	ret = qm_mb(qm, QM_MB_CMD_SQC_VFT, 0, 0, QM_MB_OP_RD);
+	if (ret)
+		return ret;
+
+	sqc_vft = io_read64(qm->io_base + QM_MAILBOX_DATA_ADDR_L);
+	*base = (sqc_vft >> QM_SQC_VFT_START_SQN_SHIFT) & QM_SQC_VFT_BASE_MASK;
+	*num = ((sqc_vft >> QM_SQC_VFT_SQ_NUM_SHIFT) & QM_SQC_VFT_NUM_MASK) + 1;
+
+	return HISI_QM_DRVCRYPT_NO_ERR;
+}
+
+static void qp_memory_uninit(struct hisi_qm *qm, uint32_t id)
+{
+	struct hisi_qp *qp = &qm->qp_array[id];
+
+	free(qp->sqe);
+	free(qp->cqe);
+}
+
+static enum hisi_drv_status qp_memory_init(struct hisi_qm *qm, uint32_t id)
```

Comment:
I suggest `s/qp_memory_init/qp_alloc/` and `s/qp_memory_uninit/qp_free/` because the names are simpler and better reflect the fact that you are indeed allocating/freeing from the heap, not just initializing/clearing some data.